executor/point_get_test.go

```diff
@@ -507,6 +507,7 @@ func (s *testPointGetSuite) TestSelectCheckVisibility(c *C) {
 		_, err = session.ResultSetToStringSlice(context.Background(), tk.Se, re)
 		c.Assert(err, NotNil)
 		c.Assert(expectErr.Equal(err), IsTrue)
+		c.Assert(re.Close(), IsNil)
```

Comment:
Though it is not elegant, `ResultSetToStringSlice` will close the result. You don't need to do it again. I think you can add this method to the interface of recordSet somehow.
 
Like making a wrapper struct having methods `ToString()` and `ToStringSlice()` implementing `RecordSet`, which based on a real recordSet implementation implementing `BaseRecordSet`.

File another issue if you are interested.

Reply:
I see, thanks for pointing out for me. Removed the redundant close,  I'll open another issue for this later.